src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -912,6 +912,39 @@ bool layout_optimizer::is_mixed_layout(program_node& prev, program_node& next, b
     return false;
 }
 
+void layout_optimizer::set_onednn_dyn_conv_preferred_format(convolution_node& node) {
+    auto input_layout = node.get_input_layout(0);
+    auto output_layout = node.get_output_layout();
+    bool i8_u8_input = input_layout.data_type == data_types::u8 || input_layout.data_type == data_types::i8;
+    bool is_fp16_input = input_layout.data_type == data_types::f16;
+    if (i8_u8_input) {
+        node.set_preferred_input_fmt(0, cldnn::format::b_fs_yx_fsv32);
+        bool i8_u8_output = output_layout.data_type == data_types::u8 || output_layout.data_type == data_types::i8;
+        if (i8_u8_output) {
+            node.set_preferred_output_fmt(0, cldnn::format::b_fs_yx_fsv32);
+        } else {
+            node.set_preferred_output_fmt(0, cldnn::format::b_fs_yx_fsv16);
+        }
+        // shallow channel
+        if (input_layout.get_partial_shape()[1].is_static() && input_layout.get_partial_shape()[1].get_length() <= 16) {
+            node.set_preferred_input_fmt(0, cldnn::format::byxf);
+        }
+        if (output_layout.get_partial_shape()[1].is_static() && output_layout.get_partial_shape()[1].get_length() <= 16) {
+            node.set_preferred_output_fmt(0, cldnn::format::byxf);
+        }
+    } else if (is_fp16_input) {
+        if (input_layout.get_partial_shape().size() <= 4)
+            node.set_preferred_output_fmt(0, format::b_fs_yx_fsv16);
+        else if (input_layout.get_partial_shape().size() == 5)
+            node.set_preferred_output_fmt(0, format::b_fs_zyx_fsv16);
+        // Use planar format for dynamic convolution with small input channel(IC <= 3)
+        if (node.is_dynamic() && input_layout.get_partial_shape()[1].is_static() &&
```

Comment:
I think `node.is_dynamic()` is a duplicated check here. Instead, I suggest to add assertion for `node.is_dynamic()` at the beginning of this function so that it is clearly stated that this is only for dynamic shape model.